wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DockerimageToolApplier.java

```diff
@@ -0,0 +1,127 @@
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.devfile.server;
+
+import static com.google.common.base.Strings.*;
+import static org.eclipse.che.api.core.model.workspace.config.Command.MACHINE_NAME_ATTRIBUTE;
+import static org.eclipse.che.api.core.model.workspace.config.MachineConfig.MEMORY_LIMIT_ATTRIBUTE;
+import static org.eclipse.che.api.devfile.server.Constants.DOCKERIMAGE_TOOL_TYPE;
+import static org.eclipse.che.api.workspace.shared.Constants.PROJECTS_VOLUME_NAME;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.devfile.model.Devfile;
+import org.eclipse.che.api.devfile.model.Endpoint;
+import org.eclipse.che.api.devfile.model.Tool;
+import org.eclipse.che.api.workspace.server.model.impl.EnvironmentImpl;
+import org.eclipse.che.api.workspace.server.model.impl.MachineConfigImpl;
+import org.eclipse.che.api.workspace.server.model.impl.RecipeImpl;
+import org.eclipse.che.api.workspace.server.model.impl.ServerConfigImpl;
+import org.eclipse.che.api.workspace.server.model.impl.VolumeImpl;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceConfigImpl;
+import org.eclipse.che.workspace.infrastructure.kubernetes.util.KubernetesSize;
+
+/**
+ * Applies Kubernetes tool configuration on provided {@link Devfile} and {@link
+ * WorkspaceConfigImpl}.
+ *
+ * @author Sergii Leshchenko
+ */
+public class DockerimageToolApplier {
+
+  private static final String DOCKERIMAGE_RECIPE_TYPE = "dockerimage";
+
+  private final String projectFolderPath;
+
+  @Inject
+  public DockerimageToolApplier(@Named("che.workspace.projects.storage") String projectFolderPath) {
+    this.projectFolderPath = projectFolderPath;
+  }
+
+  /**
+   * Applies Kubernetes tool configuration on provided {@link Devfile} and {@link
+   * WorkspaceConfigImpl}.
+   *
+   * <p>It includes:
+   *
+   * <ul>
+   *   <li>provisioning environment based on specified configuration in tool: dockerimage, env,
+   *       volumes, etc.;
+   *   <li>provisioning machine name attribute to commands that are configured to be run in the
+   *       specified tool;
+   * </ul>
+   *
+   * @param tool dockerimage tool that should be applied to devfile and workspace config
+   * @param devfile devfile that should be changed according to the provided tool
+   * @param workspaceConfig workspace config that should be changed according to the provided tool
+   * @throws IllegalArgumentException when wrong type tool is passed
+   */
+  public void apply(Tool tool, Devfile devfile, WorkspaceConfigImpl workspaceConfig) {
+    if (!DOCKERIMAGE_TOOL_TYPE.equals(tool.getType())) {
```

Comment:
would you like to add `checkArgument()` methods as it done in K8S applier ?

Reply:
I think it's OK to leave it as is since it is used once.
And ideally, it would be good to have a util class in somewhere in `commons` that would make `checkArgument` and we won't copy/paste it everywhere

Reply:
I thought that you mean usage of private method `checkArgument` but you meant method from `Preconditions` class. Good catch. Fixed